### PR TITLE
feat(profile): user info section

### PR DESCRIPTION
**Pull request by zerubeus**
# بسم الله الرحمن الرحيم

## What's new in this PR?

fixes

لله الحمد والمنة

Screenshot 2020-05-09 at 18 22 56
Screenshot 2020-05-09 at 18 23 03

### CODE REVIEW

File: src/components/modules/Profile/Profile.tsx [hunk 1]
```diff
@@ -42,7 +68,20 @@ const NEW_USER_QUERY = gql`
   }
 `
 
-const Profile: React.FC<PropType> = () => {
+const Profile: React.FC<PropType> = ({ me }) => {
+  const [isEditProfile, setEditProfile] = useState(false)
+  const classes = useStyles()
+  const currentUserId = useMemo(() => _get(me, 'uid'), [me])
+  const { loading: queryLoading, error: queryError, data: queryData } = useQuery(USER_QUERY, {
```
Comment (abumalick) [inline src/components/modules/Profile/Profile.tsx]: you can use `useQuery<MeQuery, MeQueryVariables>(` here to have the variables type checked and the data types automatically. You can import these from hasuraTypes.ts

File: src/components/modules/Profile/Profile.tsx [hunk 2]
```diff
@@ -42,7 +68,20 @@ const NEW_USER_QUERY = gql`
...
+    variables: { id: currentUserId },
+  })
+
+  const meData: TApi_Users = _get(queryData, 'users.[0]')
+
+  const userFullName = `${_upperFirst(_get(meData, 'last_name'))}  ${_upperFirst(_get(meData, 'first_name'))}`
+  const email = _get(meData, 'email')
+  const country = _upperFirst(_get(meData, 'country'))
```
Comment (abumalick) [inline src/components/modules/Profile/Profile.tsx:87]: Why don't you:
`const {country, email, last_name, first_name} = meData?.users?.[0] || {}`
?

**Review by abumalick — COMMENTED** [review]
Good work 👍

File: src/components/modules/Profile/Profile.tsx (hunk 2 above)
Comment (zerubeus) [reply src/components/modules/Profile/Profile.tsx:87]: It is preferable to use get, get is safer declarative and for your example you could do it with get as follow : 
```js
_gat(meData, 'users.[0]', {})
```

**Pull request merged**